docs/animation/armature.md

```diff
@@ -0,0 +1,183 @@
+            "samples": [0, 0.6875, 0]
+          }
+        ],
+        "cube": [
+          {
+            "variable": "offset_x",
+            "type": "uniform",
+            "interpolation": "linear",
+            "samples": [0, 0.6875, 0]
+          },
+          {
+            "variable": "axis_z",
+            "type": "uniform",
+            "interpolation": "nearest",
+            "samples": [ 1 ]
+          },
+          {
+            "variable": "origin_x",
+            "type": "uniform",
+            "interpolation": "nearest",
+            "samples": [ 0.15625 ]
+          },
+          {
+            "variable": "origin_y",
+            "type": "uniform",
+            "interpolation": "nearest",
+            "samples": [ 0.40625 ]
+          },
+          {
+            "variable": "angle",
+            "type": "uniform",
+            "interpolation": "linear",
+            "samples": [0, 120, 240, 0, 120, 240]
+          }
+        ]
+      },
+      "events": {}
+    }
+  }
+}
+
+```
+
+The file is organized in two sections, joints and clips.
+
+Joints
+--------
+Each joint defines a connection to the model file. A joint can animate any number of elements in the model file, but they are all animated with the same input value and transforms
+
+The format is like this:
+```javascript
+ {
+    "joints": {
+        <joint>, ...
+    }
+}
+    
+---
+
+<joint> ::= {
+    <string>: {  // joint name
+        <joint_definition>, ...
+    }
+}
+
+<joint_definition> ::= {
+    <string>: [ <float> ] // index_model, joint_weight
+}
+
+```
+
+- `joint_name` is the name of the joint
+- `index_model` is a 0-indexed number denoting a model element this joint controls. Must be a string (see example)
+- `joint_weight` is a weight (0-1) of how much this element will animate. The transformation matrix caused by a clip animating this element will be multiplied by this number.
+
+!!! note
+    
+    In most cases, `joint_weight` should be set to 1.0, unless you really have a reason otherwise
+
+Not all elements need to have a joint, only the ones you are animating.
+
+
+Clips
+-------
+
+Clips are essentially instructions on how to use a value to animate some collection of joints.
+They also include events to fire at certain points.
+
+They are formatted like this:
+```javascript
+
+{
+    "clips": {
+        "clip_name": {
+            "loop": <true/false>,
+            "joint_clips": {
+                <joint_clip_list>, ...
+            },
+            "events": {
+                <event> ...
+            }
+            
+        }
+    }
+}
+
+-------
+
+<joint_clip_list> ::= {
+    "joint_name": [
+        <joint_clip>, ...
+    ]
+}
+
+<joint_clip> ::= {
+    "variable": <variable>,
+    "type": "uniform",
+    "interpolation": <interpolation>,
+    "samples": [ float, ... ]
+}
+
+
+```
+
+- loop: whether or not the parameter value can continue increasing and the animation loops, or if it stops at the ending state
```

Comment:
not sure what "parameter value can continue increasing and the animation loops" means, the grammar isn't quite clear